data/src/test/java/org/akvo/flow/data/repository/DataPointDataRepositoryTest.kt

```diff
@@ -19,150 +19,126 @@
 
-    @Mock
-    internal var mockDeviceHelper = DeviceHelper(null)
+    @MockK
+    lateinit var mockMapper: DataPointImageMapper
 
-    @Mock
-    internal var mockDataSourceFactory: DataSourceFactory =
-        DataSourceFactory(null, null, null, null, null, null)
+    @MockK
+    lateinit var mediaHelper: MediaHelper
 
-    @Mock
-    internal var mediaHelper = MediaHelper()
+    @RelaxedMockK
+    lateinit var mockDatabaseDataSource: DatabaseDataSource
 
-    @Mock
-    internal var mapper = DataPointImageMapper(mediaHelper)
+    @MockK
+    lateinit var spyHttpException: HttpException
 
-    @Mock
-    internal var mockS3RestApi: S3RestApi = S3RestApi(
-        RestServiceFactory(null, null),
-        AmazonAuthHelper(null, null),
-        DateFormat.getDateInstance(),
-        BodyCreator(),
-        ""
-    )
-
-    private lateinit var spyHttpException: HttpException
-
-    private lateinit var spyRestApi: RestApi
+    lateinit var repository: DataPointDataRepository
 
     @Before
     fun setUp() {
-        `when`(mapper.getImagesList(anyList())).thenReturn(emptyList())
-        spyRestApi = spy(RestApi(mockDeviceHelper, TestRestServiceFactory(), "", ""))
-        spyHttpException = spy(HttpException(retrofit2.Response.success("")))
-        `when`(mockDataSourceFactory.dataBaseDataSource).thenReturn(mockDatabaseDataSource)
-    }
-
-    @Test(expected = AssignmentRequiredException::class)
-    fun downloadDataPointsShouldReturnExpectedErrorWhenAssignmentMissing() = runBlockingTest {
-        doThrow(spyHttpException).`when`(spyRestApi).downloadDataPoints(anyLong())
-        doReturn(HttpURLConnection.HTTP_FORBIDDEN).`when`(spyHttpException).code()
-
-        val repository =
+        MockKAnnotations.init(this)
+        every { mockMapper.getImagesList(any()) } returns emptyList()
+        every { mockDataSourceFactory.dataBaseDataSource } returns mockDatabaseDataSource
+        every { mockDatabaseDataSource.getDataPointCursor(any()) } returns null
+        repository =
             DataPointDataRepository(
                 mockDataSourceFactory,
-                spyRestApi,
+                mockRestApi,
                 mockS3RestApi,
-                mapper,
+                mockMapper,
                 mediaHelper
             )
-
-        repository.downloadDataPoints(123L)
     }
 
-    @Test(expected = HttpException::class)
-    fun downloadDataPointsShouldReturnExpectedErrorWhenNotAssignmentMissing() = runBlockingTest {
-        doThrow(spyHttpException).`when`(spyRestApi).downloadDataPoints(anyLong())
-        doReturn(HttpURLConnection.HTTP_BAD_GATEWAY).`when`(spyHttpException).code()
-
-        val repository =
-            DataPointDataRepository(
-                mockDataSourceFactory,
-                spyRestApi,
-                mockS3RestApi,
-                mapper,
-                mediaHelper
+    @Test
+    fun downloadDataPointsShouldReturnExpectedErrorWhenAssignmentMissing() = runBlocking {
```

Comment:
when assignment missing the returned code is a `403`? shouldnt it be `404`?

Reply:
oh right it's 404, I though it was 403 for some reason :thinking:

Reply:
let me cross check that in the backend as well 👍

Reply:
a `403` is returned when no `surveyId` is provided.  It should also probably be something like bad request `401` instead.  Is that a very big change for you?